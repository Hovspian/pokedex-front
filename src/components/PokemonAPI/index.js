const baseURL = 'https://pokedex-backend-server.herokuapp.com';

export const getRangeOfPokemon = (id, range) =>
  fetch(`${baseURL}/pokemon?id=${id}&range=${range}`)
    .then(res => res.json())
    .then(json => json.pokemon)

