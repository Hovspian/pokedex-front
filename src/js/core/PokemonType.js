import React from 'react';
import PropTypes from 'prop-types';
import { Badge, Popover, PopoverBody } from 'reactstrap';

import { TYPE_LOOKUP } from '../core/Constants';
import '../../styles/core/PokemonType.css';

const propTypes = {
  types: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.arrayOf(
      PropTypes.shape({
        type: PropTypes.string,
        multiplier: PropTypes.string
      }))
  ]).isRequired,
  isCheckbox: PropTypes.bool,
  toggleType: PropTypes.func,
  isWeakness: PropTypes.bool,
  isLarge: PropTypes.bool,
  checkedTypes: PropTypes.arrayOf(PropTypes.bool)
};

class PokemonType extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    };
    this.toggleHover = this.toggleHover.bind(this);
  }

  toggleHover() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  /**
   * Returns an element for the weakness with a popover if the multiplier is extra strong.
   * A multiplier is normally 2x
   *
   * @param {Object} weakness - the weakness being rendered
   * @param {String} weakness.type - the weakness's name
   * @param {String} weakness.multiplier - the weakness's multiplier
   */
  renderWithHover(weakness) {
    const multiplier = parseFloat(weakness.multiplier);
    if (multiplier > 2) {
      return (
        <span>
          {weakness.type}
          <span className="extra-damage"
                id={`extra-damage-${weakness.type}`}
                onMouseOver={this.toggleHover}
                onMouseOut={this.toggleHover}
          >
            *
          </span>
          <Popover placement="top"
                   isOpen={this.state.isOpen}
                   target={`extra-damage-${weakness.type}`}
                   hideArrow={true}
          >
            <PopoverBody>Deals {multiplier}x damage</PopoverBody>
          </Popover>
        </span>
      );
    } else {
      return <span>{weakness.type}</span>;
    }
  }

  shouldRenderChecked(checked, typeName) {
    if (!checked || !checked[TYPE_LOOKUP[typeName] - 1]) {
      return '';
    } else {
      return ' checked';
    }
  }

  shouldToggle(typeName) {
    if (this.props.isCheckbox) {
      this.props.toggleType(TYPE_LOOKUP[typeName] - 1, this.props.isWeakness);
    }
  }

  renderType(isLarge, isWeakness, type) {
    if (isLarge && isWeakness) {
      return (
        <h4>{this.renderWithHover(type)}</h4>
      );
    } else if (isLarge) {
      return (
        <h4>{type}</h4>
      );
    } else {
      return (
        <span>{type}</span>
      );
    }
  }

  render() {
    return (
      <div>
        {this.props.types.map((type, index) => {
          const typeName = this.props.isWeakness ? type.type : type;
          const typeClass = this.props.isLarge ? ' large-type' : '';
          const typeCheckbox = this.props.isCheckbox ? ' checkbox' : '';
          const typeChecked = this.shouldRenderChecked(this.props.checkedTypes, typeName);
          return (
            <Badge className={`type ${typeName}${typeClass}${typeCheckbox}${typeChecked}`}
                   key={typeName + index}
                   onClick={() => {this.shouldToggle(typeName)}}
                   pill
            >
              {this.renderType(this.props.isLarge, this.props.isWeakness, type)}
            </Badge>
          );
        })}
      </div>
    );
  }
}

PokemonType.propTypes = propTypes;

export default PokemonType;
