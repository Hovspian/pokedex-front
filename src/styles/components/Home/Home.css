.cardContainer {
  width: 100%;
  justify-content: space-between;
  display: flex;
  flex-wrap: wrap;
  padding: 20px;
}