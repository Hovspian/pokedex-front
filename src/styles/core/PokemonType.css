.type {
  margin: 3px;
}

/* Colors taken from https://bulbapedia.bulbagarden.net/wiki/Type */
.bug {
  background-color: #A8B820;
}

.electric {
  background-color: #F8D030;
}

.dark {
  background-color: #705848;
}

.dragon {
  background-color: #7038F8;
}

.fairy {
  background-color: #EE99AC;
}

.fighting {
  background-color: #C03028;
}

.fire {
  background-color: #f08030;
}

.flying {
  background-color: #A890F0;
}

.ghost {
  background-color: #705898;
}

.grass {
  background-color: #78C850;
}

.ground {
  background-color: #E0C068;
}

.ice {
  background-color: #98D8D8;
}

.normal {
  background-color: #a8a878;
}

.poison {
  background-color: #A040A0;
}

.psychic {
  background-color: #F85888;
}

.rock {
  background-color: #B8A038;
}

.steel {
  background-color: #B8B8D0;
}

.water {
  background-color: #6890F0;
}

.bug-border {
  color: #A8B820;
}

.electric-border {
  color: #F8D030;
}

.dark-border {
  color: #705848;
}

.dragon-border {
  color: #7038F8;
}

.fairy-border {
  color: #EE99AC;
}

.fighting-border {
  color: #C03028;
}

.fire-border {
  color: #f08030;
}

.flying-border {
  color: #A890F0;
}

.ghost-border {
  color: #705898;
}

.grass-border {
  color: #78C850;
}

.ground-border {
  color: #E0C068;
}

.ice-border {
  color: #98D8D8;
}

.normal-border {
  color: #a8a878;
}

.poison-border {
  color: #A040A0;
}

.psychic-border {
  color: #F85888;
}

.rock-border {
  color: #B8A038;
}

.steel-border {
  color: #B8B8D0;
}

.water-border {
  color: #6890F0;
}

.large-type {
  width: 130px;
}

.checked {
  box-shadow: 0px 0px 10px 10px #ddd;
}

.checkbox {
  margin: 10px 0;
  cursor: pointer;
  user-select: none;
}

.extra-damage {
  margin-left: 5px;
  padding: 17px 6px 5px 6px;
  display: inline-block;
  border-radius: 50%;
  background-color: grey;
  vertical-align: sub;
  line-height: 0;
  position: relative;
  cursor: pointer;
}
