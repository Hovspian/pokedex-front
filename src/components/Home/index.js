import React, { Component } from 'react';
import Header from '../Header';
import PokemonCard from '../PokemonCard';
import NoPokemon from '../NoPokemon';
import { getRangeOfPokemon } from '../PokemonAPI';

import '../../styles/components/Home/Home.css'

class Home extends Component {

  constructor() {
    super();
    this.state = {
      pokemon: []
    }
  }

  componentDidMount() {
    this.fetchPokemon(1, 20);
  }

  fetchPokemon(id, range) {
    getRangeOfPokemon(id, range)
      .then(newPokemon => {
        const oldPokemon = this.state.pokemon.slice();
        this.setState({
          pokemon: [...oldPokemon, ...newPokemon]
        });
      })

      // TODO: Need to do something if error on fetch from server
      .catch(error => {
        window.console.log('Unable to fetch Pokeon data from server with error: ' + error);
      });
  }

  displayPokemonCards() {
    if (this.state.pokemon.length === 0) {
      return <NoPokemon />
    }
    return this.state.pokemon.map(pokemon => {
      return <PokemonCard {...pokemon} key={pokemon.id} />
    });
  }

  render() {
    return (
      <div>
        <Header />
        <div className="cardContainer">
          {this.displayPokemonCards()}
        </div>
      </div>
    )
  }
}

export default Home;