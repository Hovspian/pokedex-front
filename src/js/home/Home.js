import React from 'react';
import InfiniteScroll from 'react-infinite-scroller';
import { Button } from 'reactstrap';

import { MAX_POKEMON } from '../core/Constants';
import { isMobile } from '../core/MobileCheck';
import { getRangeOfPokemon, getPokemonDetails } from '../core/PokemonAPI';
import Header from './Header';
import PokemonCard from './PokemonCard';
import Loader from './Loader';
import LoadingError from './LoadingError';
import PokemonModal from '../modal/PokemonModal';
import Search from './Search';
import EmptySearch from './EmptySearch';

import '../../styles/home/Home.css';

class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      pokemon: [],
      pokemonDetails: null,
      error: '',
      shouldInfiniteScroll: false,
      initialLoad: true,
      modal: false,
      hasMore: true,
      selectedForm: 0,
      isMobile: false,
    };
    this.perPage = 20;

    this.getDetails = this.getDetails.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
    this.enableInfiniteScroll = this.enableInfiniteScroll.bind(this);
    this.selectForm = this.selectForm.bind(this);
    this.fetchPokemon = this.fetchPokemon.bind(this);
    this.setMobileState = this.setMobileState.bind(this);
  }

  componentDidMount() {
    this.setMobileState();
    window.addEventListener("resize", this.setMobileState);
  }

  /**
   * Fetches a range of pokemon from the server based on search criteria.
   * Does some validation and parses Types arrays for checked types
   * if appendPokemon, then append results to current state, else replace
   * @param  {Object} options - search options to be fetched
   * @param  {boolean} appendPokemon - should append results
   * @return {void}
   */
  fetchPokemon(options, appendPokemon) {
    if (this.state.error || !options) {
      return;
    }

    const searchOptions = {};

    // pushes index + 1 because Type array starts at 0
    // but first Type in lookup is 1 (Normal)
    if (options.hasOwnProperty('selectedTypes')) {
      const searchTypes = [];
      options.selectedTypes.forEach((isChecked, index) => {
        if (isChecked) {
          searchTypes.push(index + 1);
        }
      });
      if (searchTypes.length > 0) {
        searchOptions.types = searchTypes;
      }
    }

    // pushes index + 1 because Type array starts at 0
    // but first Type in lookup is 1 (Normal)
    if (options.hasOwnProperty('selectedWeaknesses')) {
      const searchWeaknesses = [];
      options.selectedWeaknesses.forEach((isChecked, index) => {
        if (isChecked) {
          searchWeaknesses.push(index + 1);
        }
      });
      if (searchWeaknesses.length > 0) {
        searchOptions.weaknesses = searchWeaknesses;
      }
    }

    if (options.rangeStart && options.rangeEnd) {
      searchOptions.id = options.rangeStart;
      searchOptions.range = options.rangeEnd - options.rangeStart + 1;
    }

    if (options.abilityName) {
      searchOptions.ability = options.abilityName;
    }

    // idOrName can be a string (Name) or number (ID)
    if (options.idOrName) {
      if (isNaN(options.idOrName)) {
        searchOptions.name = options.idOrName;
      } else {
        const id = parseInt(options.idOrName, 10);

        // if searching by an ID, but does not fall into the
        // range, then return empty search result
        if (id < options.rangeStart || id > options.rangeEnd) {
          this.setEmptySearch();
          return;
        }
        searchOptions.id = options.idOrName;
        searchOptions.range = 1;
      }
    }

    // if no valid search params, then just return empty search;
    if (Object.keys(searchOptions).length < 1) {
      this.setEmptySearch();
      return;
    }

    return getRangeOfPokemon(searchOptions)
      .then(newPokemon => {
        if (appendPokemon) {
          const oldPokemon = [...this.state.pokemon];
          const hasMore = searchOptions.id + searchOptions.range < MAX_POKEMON ? true : false;
          this.setState({
            pokemon: [...oldPokemon, ...newPokemon],
            initialLoad: false,
            hasMore: hasMore,
          });
        } else {
          this.setState({
            pokemon: newPokemon,
            initialLoad: false,
            hasMore: false,
            shouldInfiniteScroll: false,
          });
        }
      })
      .catch(error => {
        this.setState({
          error: error.message,
          shouldInfiniteScroll: false,
        });
      });
  }

  setEmptySearch() {
    this.setState({
      pokemon: [],
      initialLoad: false,
      hasMore: false,
      shouldInfiniteScroll: false,
    });
  }

  fetchStandardRange() {
    // returns early if does not have more to fetch
    // InfiniteScroll wants to pull one last time
    // if have scroll enabled with an empty search
    if (!this.state.hasMore) {
      return;
    }

    const rangeStart = this.state.pokemon.length + 1;
    let rangeEnd;
    if (rangeStart + this.perPage > MAX_POKEMON) {
      this.setState({ shouldInfiniteScroll: false })
      rangeEnd = MAX_POKEMON;
    } else {
      rangeEnd = rangeStart + this.perPage - 1;
    }

    this.fetchPokemon({ rangeStart, rangeEnd }, true);
  }

  displayPokemonCards() {
    return this.state.pokemon.map(pokemon => {
      return <PokemonCard {...pokemon} key={pokemon.id} handlePokemonCardClick={this.getDetails} />
    });
  }

  renderLoadButton() {
    return this.state.hasMore ? (
      <div>
        <Button
          className="load-button"
          color="primary" size="lg"
          onClick={() => {
            this.enableInfiniteScroll();
            this.clearError();
          }}
        >
          Load more Pok√©mon
        </Button>
      </div>
    ) : null;
  }

  enableInfiniteScroll() {
    this.setState({ shouldInfiniteScroll: true });
  }

  clearError() {
    this.setState({ error: '' });
  }

  /**
  * Fetches a pokemon details from server and adds them to the current state
  * @param  {int} id - the id of the pokemon to fetch
  * @return {void}
  */
  getDetails(id) {
    if (this.state.error ||
       id < 1 ||
       id > 807)
      return;

    if (this.state.pokemonDetails && this.state.pokemonDetails.id === id) {
      this.setState({
        modal: true,
        selectedForm: 0,
      });
      return;
    }

    return getPokemonDetails(id)
      .then(pokemonDetails => {
        this.setState({
          pokemonDetails,
          modal: true,
          selectedForm: 0,
        });
      })

      .catch(error => {
        this.setState({
          error: error.message,
        });
      });
  }

  selectForm(index) {
    if (!this.state.pokemonDetails || index >= this.state.pokemonDetails.forms.length || index < 0)
      return;
    this.setState({ selectedForm: index });
  }

  setMobileState() {
    this.setState({ isMobile: isMobile() });
  }

  handleCloseModal() {
    this.setState({ modal: false })
  }

  render() {
    let cards = this.displayPokemonCards();
    let loadMoreButton = !this.state.shouldInfiniteScroll ?
      this.renderLoadButton() : null;
    let error = this.state.error ? <LoadingError error={this.state.error} /> : null;
    let emptySearch = (this.state.pokemon.length === 0 && !this.state.initialLoad) ? <EmptySearch /> : null;

    return (
      <div className={`main${this.state.isMobile ? ' mobile' : ''}`} align="center">
        <PokemonModal
          handleCloseModal={this.handleCloseModal}
          modal={this.state.modal}
          getPokemonDetails={this.getDetails}
          data={this.state.pokemonDetails}
          selectForm={(index) => { this.selectForm(index) }}
          selectedForm={this.state.selectedForm}
        />
        <Header />
        <Search fetchPokemon={this.fetchPokemon} />
        <InfiniteScroll
          initialLoad={this.state.initialLoad}
          pageStart={0}
          loadMore={this.fetchStandardRange.bind(this)}
          hasMore={this.state.shouldInfiniteScroll || this.state.initialLoad}
          loader={<Loader key="loader" />}
        >
          <div className="card-container">
            {cards}
          </div>
        </InfiniteScroll>
        {loadMoreButton}
        {emptySearch}
        {error}
      </div>
    )
  }
}

export default Home;
