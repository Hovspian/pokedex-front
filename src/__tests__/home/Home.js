import React from 'react';
import { shallow } from 'enzyme';

import Home from '../../js/home/Home';

import * as PokemonAPI from '../../js/core/PokemonAPI';
import * as MobileCheck from '../../js/core/MobileCheck';
import ExampleJSON from '../../../local_data/example_detail.json';
import { TYPES } from '../../js/core/Constants';

describe('Home Component', () => {
  it('should render Home component without crashing', () => {
    shallow(<Home />);
  });

  it('should match Home component snapshot', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mobile', () => {
    beforeEach(() => {
      MobileCheck.isMobile = jest.fn();
    });

    it('should set `state.isMobile` to true when the window is mobile', () => {
      MobileCheck.isMobile.mockImplementation(() => true);
      const wrapper = shallow(<Home />);
      expect(wrapper.state('isMobile')).toBe(true);
    });

    it('should set `state.isMobile` to false when the window is not mobile', () => {
      MobileCheck.isMobile.mockImplementation(() => false);
      const wrapper = shallow(<Home />);
      expect(wrapper.state('isMobile')).toBe(false);
    });
  });

  describe('handleCloseModal', () => {
    let wrapper, instance;
    beforeEach(() => {
      wrapper = shallow(<Home />);
      instance = wrapper.instance();
    });

    it('should set `state.modal` to false when closing modal', () => {
      wrapper.setState({ modal: true });
      expect(wrapper.state('modal')).toBe(true);

      instance.handleCloseModal();
      expect(wrapper.state('modal')).toBe(false);
    });
  });

  describe('selectForm', () => {
    let wrapper, instance;
    beforeEach(() => {
      wrapper = shallow(<Home />);
      instance = wrapper.instance();

      wrapper.setState({
        pokemonDetails: ExampleJSON,
        selectedForm: 0,
      });
    });

    it('should set state.selectedForm to the passed index when the index is valid', () => {
      wrapper.setState({ selectedForm: 3 });
      expect(wrapper.state('selectedForm')).toBe(3);

      instance.selectForm(0);
      expect(wrapper.state('selectedForm')).toBe(0);
    });

    it('should do nothing when the passed index is out of range of the forms', () => {
      expect(wrapper.state('selectedForm')).toBe(0);

      instance.selectForm(10);
      expect(wrapper.state('selectedForm')).toBe(0);
    });

    it('should do nothing when the passed index is negative', () => {
      expect(wrapper.state('selectedForm')).toBe(0);

      instance.selectForm(-1);
      expect(wrapper.state('selectedForm')).toBe(0);
    });

    it('should do nothing when the state does not have pokemonDetails', () => {
      wrapper.setState({ pokemonDetails: null });
      expect(wrapper.state('selectedForm')).toBe(0);

      instance.selectForm(1);
      expect(wrapper.state('selectedForm')).toBe(0);
    });

  });

  describe('Test API is being called', () => {
    let wrapper, instance;
    beforeEach(() => {
      wrapper = shallow(<Home />);
      instance = wrapper.instance();
    });

    describe('getPokemonDetails', () => {
      beforeEach(() => {
        PokemonAPI.getPokemonDetails = jest.fn();
        PokemonAPI.getPokemonDetails.mockImplementation(() => Promise.resolve(ExampleJSON));
      });

      it('should call API if clicked on pokemon card', () => {
        return instance.getDetails(1)
          .then(() => {
            expect(PokemonAPI.getPokemonDetails).toHaveBeenCalledTimes(1);
            expect(instance.state.modal).toBe(true);
            expect(instance.state.pokemonDetails).toEqual(ExampleJSON);
            expect(instance.state.selectedForm).toBe(0);
          });
      });

      it('should not call API when state already contains details', () => {
        wrapper.setState({
          pokemonDetails: ExampleJSON,
          modal: false,
          selectedForm: 2,
        });

        instance.getDetails(12);
        expect(PokemonAPI.getPokemonDetails).toHaveBeenCalledTimes(0);
        expect(instance.state.modal).toBe(true);
        expect(instance.state.pokemonDetails).toEqual(ExampleJSON);
        expect(instance.state.selectedForm).toBe(0);
      });
    });

    it('should call API when fetching pokemon', () => {
      PokemonAPI.getRangeOfPokemon = jest.fn();
      PokemonAPI.getRangeOfPokemon.mockImplementation(() => Promise.resolve([{
        'id': 1,
        'name': 'Bulbasaur',
        'types': [
          'grass',
          'poison'
        ],
        'image_path': '/sprites/pokemon/large/1.png',
      }]));

      return instance.fetchPokemon({ rangeStart: 1, rangeEnd: 1 }, false)
        .then(() => {
          expect(PokemonAPI.getRangeOfPokemon).toHaveBeenCalledTimes(1);
          expect(instance.state.pokemon).toEqual(
            expect.arrayContaining([
              expect.objectContaining({
                'id': 1,
                'name': 'Bulbasaur',
                'types': [
                  'grass',
                  'poison'
                ],
                'image_path': '/sprites/pokemon/large/1.png',
              })
            ])
          );
        });
    });

    it('should not call API if have no search options', () => {
      PokemonAPI.getRangeOfPokemon = jest.fn();

      instance.fetchPokemon();
      expect(PokemonAPI.getRangeOfPokemon).toHaveBeenCalledTimes(0);
      expect(instance.state.pokemon).toEqual([]);

    });

    it('should create a correct search object from given params', () => {
      const validSearchParams = {
        rangeStart: 1,
        rangeEnd: 500,
        selectedTypes: new Array(TYPES.length).fill(true),
        selectedWeaknesses: new Array(TYPES.length).fill(true)
      };

      const bulbReturn = [{
        'id': 1,
        'name': 'Bulbasaur',
        'types': [
          'grass',
          'poison'
        ],
        'image_path': '/sprites/pokemon/large/1.png',
      }];

      const expectedSearch = {
        id: 1,
        range: 500,
        types: Array.from(Array(TYPES.length), (_, x) => x + 1),
        weaknesses: Array.from(Array(TYPES.length), (_, x) => x + 1),
      }

      PokemonAPI.getRangeOfPokemon = jest.fn();
      PokemonAPI.getRangeOfPokemon.mockImplementation(() => Promise.resolve(bulbReturn));

      return instance.fetchPokemon(validSearchParams, false)
        .then(() => {
          expect(PokemonAPI.getRangeOfPokemon).toHaveBeenCalledWith(expectedSearch);
          expect(instance.state.pokemon).toEqual(bulbReturn);
        });
    });
  });
});
