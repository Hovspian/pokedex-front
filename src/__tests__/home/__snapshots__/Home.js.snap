// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match Home component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "align": "center",
      "children": Array [
        <PokemonModal
          data={null}
          getPokemonDetails={[Function]}
          handleCloseModal={[Function]}
          isMobile={false}
          modal={false}
          selectForm={[Function]}
          selectedForm={0}
/>,
        <Header />,
        <Search
          fetchPokemon={[Function]}
/>,
        <InfiniteScroll
          element="div"
          hasMore={true}
          initialLoad={true}
          isReverse={false}
          loadMore={[Function]}
          loader={<Loader />}
          pageStart={0}
          ref={null}
          threshold={250}
          useCapture={false}
          useWindow={true}
>
          <div
                    className="card-container"
          >
                    
          </div>
</InfiniteScroll>,
        <div>
          <Button
                    className="load-button"
                    color="primary"
                    onClick={[Function]}
                    size="lg"
                    tag="button"
          >
                    Load more Pokémon
          </Button>
</div>,
        null,
        null,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": null,
          "getPokemonDetails": [Function],
          "handleCloseModal": [Function],
          "isMobile": false,
          "modal": false,
          "selectForm": [Function],
          "selectedForm": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fetchPokemon": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="card-container"
>
            
</div>,
          "element": "div",
          "hasMore": true,
          "initialLoad": true,
          "isReverse": false,
          "loadMore": [Function],
          "loader": <Loader />,
          "pageStart": 0,
          "ref": null,
          "threshold": 250,
          "useCapture": false,
          "useWindow": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            className="load-button"
            color="primary"
            onClick={[Function]}
            size="lg"
            tag="button"
>
            Load more Pokémon
</Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Load more Pokémon",
            "className": "load-button",
            "color": "primary",
            "onClick": [Function],
            "size": "lg",
            "tag": "button",
          },
          "ref": null,
          "rendered": "Load more Pokémon",
          "type": [Function],
        },
        "type": "div",
      },
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "align": "center",
        "children": Array [
          <PokemonModal
            data={null}
            getPokemonDetails={[Function]}
            handleCloseModal={[Function]}
            isMobile={false}
            modal={false}
            selectForm={[Function]}
            selectedForm={0}
/>,
          <Header />,
          <Search
            fetchPokemon={[Function]}
/>,
          <InfiniteScroll
            element="div"
            hasMore={true}
            initialLoad={true}
            isReverse={false}
            loadMore={[Function]}
            loader={<Loader />}
            pageStart={0}
            ref={null}
            threshold={250}
            useCapture={false}
            useWindow={true}
>
            <div
                        className="card-container"
            >
                        
            </div>
</InfiniteScroll>,
          <div>
            <Button
                        className="load-button"
                        color="primary"
                        onClick={[Function]}
                        size="lg"
                        tag="button"
            >
                        Load more Pokémon
            </Button>
</div>,
          null,
          null,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": null,
            "getPokemonDetails": [Function],
            "handleCloseModal": [Function],
            "isMobile": false,
            "modal": false,
            "selectForm": [Function],
            "selectedForm": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fetchPokemon": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="card-container"
>
              
</div>,
            "element": "div",
            "hasMore": true,
            "initialLoad": true,
            "isReverse": false,
            "loadMore": [Function],
            "loader": <Loader />,
            "pageStart": 0,
            "ref": null,
            "threshold": 250,
            "useCapture": false,
            "useWindow": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "card-container",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              className="load-button"
              color="primary"
              onClick={[Function]}
              size="lg"
              tag="button"
>
              Load more Pokémon
</Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Load more Pokémon",
              "className": "load-button",
              "color": "primary",
              "onClick": [Function],
              "size": "lg",
              "tag": "button",
            },
            "ref": null,
            "rendered": "Load more Pokémon",
            "type": [Function],
          },
          "type": "div",
        },
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
